packages/dds/map/src/test/map.spec.ts

```diff
@@ -456,5 +456,113 @@ describe("Map", () => {
                 assert.equal(await waitP2, "resolved", "promise not resolved after key is available in remote map");
             });
         });
+
+        describe("Garbage Collection", () => {
+            it("can generate GC nodes with handles in data", () => {
+                const subMap = factory.create(dataStoreRuntime, "subMap");
+                map.set("object", subMap.handle);
+
+                containerRuntimeFactory.processAllMessages();
+
+                // Verify the GC nodes returned by summarize.
+                const gcNodes = map2.summarize().gcNodes;
+                assert.strictEqual(gcNodes.length, 1, "There should only be one GC node in summary");
+                assert.strictEqual(gcNodes[0].id, "/", "GC node's id should be /");
+                assert.deepStrictEqual(
+                    gcNodes[0].outboundRoutes,
+                    [subMap.handle.absolutePath],
+                    "GC node's outbound routes is incorrect");
+            });
+
+            it("can generate GC nodes when handles are removed from data", () => {
```

Comment:
Is there any way to reuse logic across different DDSs?
It feels like you can build an interface that does various common things (like add one reference, add multiple reference to same thing, add 2 references, summarize) and then have each DDS test code implement such interface and have "core" / "test driver" to be shared across all of them somewhere in shared test code.
Would it be useful?

Reply:
I think it might be useful. It would be convenient to add more tests that apply to all these DDSs.
I thought of something similar but didn't do it because I wasn't sure if wanted to share test code across DDS (There is no such code today and sharing is limited to a DDS package).
But it makes sense to share because its a lot of duplicate code. I will probably give it a try and see how we feel about them.

Reply:
@markfields just is adding some shared test code in #4422. could be a good landing place

Reply:
I started adding it to a "fluid-internal/test-dds-utils" package under "packages/dds". I don't want to leak it outside the dds package.

Reply:
I'd lean towards `test-runtime-utils` rather than a new package. This would be a fine fit there, I think - this package already has a dependency on it.

Reply:
The utils that I am adding won't be used outside of the packages/dds directory. Also, test-runtime-utils is not an internal package and I would not prefer to add utilities to it that won't be used externally.

Reply:
However, if we decide that these utils should be used outside the package and / or it should be exposed externally, it can be moved to test-runtime-utils.